### DISCUSSION TITLE
Custom JavaScript Action Runner Support for Bun as a Runtime

### DISCUSSION THREAD

**Discussion by dwisiswant0** [actions/runner#2846] [Ideas]
## Summary

I propose the addition of a custom JavaScript action runner for GitHub Actions that supports [Bun](https://bun.sh/) as a runtime. This would enable developers to seamlessly use Bun for running, building, testing, and debugging JavaScript and TypeScript actions within the GitHub Actions workflow. While I may be in the wrong place to suggest this idea, I believe that this repository is more appropriate than GitHub Actions Toolkit, [actions/toolkit](https://github.com/actions/toolkit), for such a proposal.

## Description

GitHub Actions currently supports the Node.js runtime exclusively for running JavaScript actions. To increase flexibility and better serve the JavaScript and TypeScript developer community, it would be advantageous to introduce a custom JavaScript action runner that includes support for Bun as a runtime option. Here's a detailed explanation of how this feature could be implemented:

```yaml
# action.yml
runs:
  using: 'bun1.0' # or 'bun1', or 'bun1.0.1'
  main: 'index.js'
```

## Conclusion

Introducing a custom JavaScript action runner for Bun would empower JavaScript and TypeScript devs to harness Bun's capabilities within GitHub Actions. This addition could streamline the workflow for building, testing, and deploying apps, ultimately improving the overall experience of using GitHub Actions for JavaScript and TypeScript projects.

**Comment by aymenha** [comment]
This might be obvious for some, but IMO the primary advantage Bun provides over Node.js as a runtime is its native support for TypeScript and the Auto-install feature, which eliminates the need for dealing with `node_modules` or bundling a compiled js file, just to name a few. I'm pretty sure that many users would see the usability of such an addition.

**Comment by nebuk89** [comment]
I <3 the idea, I get the use case as well for lowering the modules/reduced complexity.

This isn't something we have the time to build out/look at doing currently (this calendar year) but if you see public contributions heading towards this, poke me so I can get reviews sorted for them :)
